pkg/kubelet/dockertools/manager_test.go

```diff
@@ -1296,49 +1298,36 @@ func TestSyncPodWithPullPolicy(t *testing.T) {
 		},
 	}
 
-	runSyncPod(t, dm, fakeDocker, pod, nil)
-
-	fakeDocker.Lock()
+	expectedStatusMap := make(map[string]*api.ContainerState)
+	expectedStatusMap["bar"] = &api.ContainerState{Waiting: nil, Running: &api.ContainerStateRunning{util.Time{}}}
+	expectedStatusMap["bar2"] = &api.ContainerState{Waiting: nil, Running: &api.ContainerStateRunning{util.Time{}}}
+	expectedStatusMap["bar3"] = &api.ContainerState{Waiting: nil, Running: &api.ContainerStateRunning{util.Time{}}}
+	expectedStatusMap["bar4"] = &api.ContainerState{Waiting: nil, Running: &api.ContainerStateRunning{util.Time{}}}
+	expectedStatusMap["bar1"] = &api.ContainerState{Waiting: &api.ContainerStateWaiting{Reason: kubecontainer.ErrImageNeverPull.Error()}, Running: nil}
 
-	eventSet := []string{
-		`Pulling Pulling image "pod_infra_image"`,
-		`Pulled Successfully pulled image "pod_infra_image"`,
-		`Pulling Pulling image "pull_always_image"`,
-		`Pulled Successfully pulled image "pull_always_image"`,
-		`Pulling Pulling image "pull_if_not_present_image"`,
-		`Pulled Successfully pulled image "pull_if_not_present_image"`,
-		`Pulled Container image "existing_one" already present on machine`,
-		`Pulled Container image "want:latest" already present on machine`,
+	runSyncPod(t, dm, fakeDocker, pod, nil)
+	statuses, err := dm.GetPodStatus(pod)
+	if err != nil {
+		t.Errorf("unable to get pod status")
 	}
-
-	recorder := dm.recorder.(*record.FakeRecorder)
-
-	var actualEvents []string
-	for _, ev := range recorder.Events {
-		if strings.HasPrefix(ev, "Pull") {
-			actualEvents = append(actualEvents, ev)
+	for _, c := range pod.Spec.Containers {
+		if containerStatus, ok := api.GetContainerStatus(statuses.ContainerStatuses, c.Name); ok {
+			assert.Equal(t, containerStatus.State.Running, expectedStatusMap[c.Name].Running, "for container %s", c.Name)
+			assert.Equal(t, containerStatus.State.Waiting, expectedStatusMap[c.Name].Waiting, "for container %s", c.Name)
 		}
 	}
-	sort.StringSlice(actualEvents).Sort()
-	sort.StringSlice(eventSet).Sort()
-	if !reflect.DeepEqual(actualEvents, eventSet) {
-		t.Errorf("Expected: %#v, Actual: %#v", eventSet, actualEvents)
-	}
 
+	fakeDocker.Lock()
```

Comment:
I think it's safe to access fakeDocker here (unrelated to your change), but locking it is fine too. Could you to `defer fakeDocker.Unlock()` Here? Thanks.